File: apps/patter-voice-sdk/libraries/typescript/src/stt/whisper.ts
/** OpenAI Whisper STT for Patter pipeline mode. */
import { WhisperSTT as _WhisperSTT, type WhisperResponseFormat } from "../providers/whisper-stt";

/** Constructor options for the Whisper `STT` adapter. */
export interface WhisperSTTOptions {
  /** API key. Falls back to OPENAI_API_KEY env var when omitted. */
  apiKey?: string;
  model?: string;
  language?: string;
  bufferSize?: number;
  /** ``"verbose_json"`` exposes segment-level confidence / timestamps. */
  responseFormat?: WhisperResponseFormat;
}

/**
 * OpenAI Whisper STT.
 *
 * @example
 * ```ts
 * import * as whisper from "getpatter/stt/whisper";
 * const stt = new whisper.STT();              // reads OPENAI_API_KEY
 * const stt = new whisper.STT({ apiKey: "sk-...", language: "en" });
 * ```
 */
export class STT extends _WhisperSTT {
  static readonly providerKey = "whisper";
  constructor(opts: WhisperSTTOptions = {}) {
    const key = opts.apiKey ?? process.env.OPENAI_API_KEY;
    if (!key) {
      throw new Error(
        "Whisper STT requires an apiKey. Pass { apiKey: 'sk-...' } or " +
          "set OPENAI_API_KEY in the environment.",
      );
    }
    super(key, opts.language, opts.model ?? "whisper-1", opts.bufferSize, opts.responseFormat ?? "json");
  }
}

File: apps/patter-voice-sdk/libraries/python/getpatter/stt/whisper.py
"""OpenAI Whisper STT for Patter pipeline mode."""

from __future__ import annotations

import os
from typing import ClassVar

from getpatter.providers.whisper_stt import WhisperSTT as _WhisperSTT

__all__ = ["STT"]


class STT(_WhisperSTT):
    """OpenAI Whisper STT (buffered HTTP transcription).

    Example::

        from getpatter.stt import whisper

        stt = whisper.STT()                 # reads OPENAI_API_KEY
        stt = whisper.STT(api_key="sk-...", language="it")
    """

    provider_key: ClassVar[str] = "whisper"

    def __init__(
        self,
        api_key: str | None = None,
        *,
        language: str = "en",
        model: str = "whisper-1",
    ) -> None:
        key = api_key or os.environ.get("OPENAI_API_KEY")
        if not key:
            raise ValueError(
                "Whisper STT requires an api_key. Pass api_key='sk-...' or "
                "set OPENAI_API_KEY in the environment."
            )
        super().__init__(api_key=key, language=language, model=model)
